[#1895] Make extended metadata url clickable

mauriel133 opened the pull request:
I implemented this as a reusable class. It can be used to make occurrences of URLs inside some element clickable.

### Example:
**To initialize on page load add the class `"url-clickable"` to your item:**
`<span class="url-clickable">Some text and a URL https://github.com/hydroshare/hydroshare/issues/898</span>`

**To initialize dynamically call the `urlClickable` method:**
`$("#mySelector").urlClickable();`

hyi commented on theme/static/js/custom.js:
```diff
@@ -1,115 +1,129 @@
+            e.preventDefault();
+        });
+        return false;
+    });
```
@mauriel133 Do we really need this ```$(document).ready(function () {``` embedded within another ```$(document).ready(function () {``` function?

mauriel133 replied:
I don't think so. This existed before I started working in this project. I can fix it.

hyi replied:
Agreed. I see it existed before. It'd be great if you can fix it now that you are at it.

hyi commented on hs_core/templates/pages/genericresource.html:
```diff
@@ -1447,7 +1452,7 @@ <h4 class="modal-title">Confirm files deletion</h4>
             t.rows(). every(function ( rowIdx, tableLoop, rowLoop ) {
                 var extra_meta_name = this.data()[0].trim();
-                var extra_meta_value = this.data()[1].trim();
+                var extra_meta_value = $("<div/>").html(this.data()[1]).text();
```
@mauriel133 It is probably better to keep ```.trim()``` for ```this.data()[1]``` to make it like ```var extra_meta_value = $("<div/>").html(this.data()[1].trim()).text();```

mauriel133 replied:
I agree.

hyi commented on hs_core/templates/pages/genericresource.html:
```diff
@@ -1357,7 +1358,7 @@ <h4 class="modal-title">Confirm files deletion</h4>
                 var row_to_edit = t.row("#" + row_id_str);
                 var oldname = row_to_edit.data()[0];
-                var oldvalue = row_to_edit.data()[1];
+                var oldvalue = $("<div/>").html(row_to_edit.data()[1]).text();
```
@mauriel133 This is related to the comment below on adding ```.trim()``` call. I am not sure whether it is needed or not. But I think we need to keep the consistency meaning either we don't add ```.trim()``` call as you have done currently, or if you add ```.trim()``` call to the other place, make sure you do the same thing here as well to keep them consistent.

mauriel133 replied:
I think it is better to have it just in case something messes up with the HTML input.

hyi commented:
@mauriel133 It looks good, just left a couple of comments for your consideration.

mauriel133 commented:
@hyi, I made the changes you've suggested and adjusted some of the style in these tabs for consistency.

I also found and fixed a bug causing errors when typing in the same URL multiple times in one entry.

hyi commented:
@mauriel133 +1 from me.
